emcc.py

```diff
@@ -3016,6 +3016,7 @@ def phase_final_emitting(options, state, target, wasm_target, memfile):
     src = read_file(final_js)
     final_js += '.esmeta.js'
     write_file(final_js, src.replace('EMSCRIPTEN$IMPORT$META', 'import.meta'))
+    shared.get_temp_files().note(final_js)
```

Comment:
What this temp file leaking previously?

Reply:
Yes, this was noted in this PR using the `EMTEST_DETECT_TEMPFILE_LEAKS=1` env variable, which is set by default in CI. I could make a separate PR for this, if you like.